DependencyInjection/Configuration.php

```diff
@@ -130,11 +130,20 @@ public function getConfigTreeBuilder()
                     ->useAttributeAsKey('id')
                     ->prototype('array')
                         ->fixXmlConfig('context')
+                        ->fixXmlConfig('template')
                         ->fixXmlConfig('setting')
                         ->children()
                             ->arrayNode('contexts')
                                 ->prototype('scalar')->end()
                             ->end()
+                            ->arrayNode('templates')
+                                ->prototype('array')
+                                    ->children()
+                                        ->scalarNode('name')->cannotBeEmpty()->end()
```

Comment:
The ``name`` is the display name for the block chooser and could contain some special chars. IMO a key couldn't have these.